libs/langchain-openai/src/azure/chat_models.ts

```diff
@@ -450,21 +474,43 @@ export class AzureChatOpenAI extends ChatOpenAI {
         openAIBasePath?: string;
         deploymentName?: string;
       } & BaseChatModelParams & {
-        configuration?: ClientOptions & LegacyOpenAIInput;
+        configuration?: ClientOptions;
       }
   ) {
-    const newFields = fields ? { ...fields } : fields;
-    if (newFields) {
-      // don't rewrite the fields if they are already set
-      newFields.azureOpenAIApiDeploymentName =
-        newFields.azureOpenAIApiDeploymentName ?? newFields.deploymentName;
-      newFields.azureOpenAIApiKey =
-        newFields.azureOpenAIApiKey ?? newFields.openAIApiKey;
-      newFields.azureOpenAIApiVersion =
-        newFields.azureOpenAIApiVersion ?? newFields.openAIApiVersion;
-    }
+    super(fields);
+    this.azureOpenAIApiKey =
+      fields?.azureOpenAIApiKey ??
+      fields?.openAIApiKey ??
+      fields?.apiKey ??
+      getEnvironmentVariable("AZURE_OPENAI_API_KEY");
+
+    this.azureOpenAIApiInstanceName =
+      fields?.azureOpenAIApiInstanceName ??
+      getEnvironmentVariable("AZURE_OPENAI_API_INSTANCE_NAME");
+
+    this.azureOpenAIApiDeploymentName =
+      fields?.azureOpenAIApiDeploymentName ??
+      fields?.deploymentName ??
+      getEnvironmentVariable("AZURE_OPENAI_API_DEPLOYMENT_NAME");
 
-    super(newFields);
+    this.azureOpenAIApiVersion =
+      fields?.azureOpenAIApiVersion ??
+      fields?.openAIApiVersion ??
+      getEnvironmentVariable("AZURE_OPENAI_API_VERSION");
```

Comment:
this feels funky - should we just update `this.openAIApiVersion`?

Reply:
WDYM?